core/src/main/java/jenkins/model/logging/LoggingMethod.java

```diff
@@ -16,32 +20,52 @@
  * @since TODO
  */
 public abstract class LoggingMethod implements Serializable {
-  
-  /**
-   * Decorates logging on the Jenkins master side.
-   * These filters can be also used for log redirection and multi-reporting.
-   * @param build Build to be decorated
-   * @return Log filter on the master. {@code null} if no custom implementation
-   */
-  @CheckForNull
-  public abstract ConsoleLogFilter createLoggerDecorator(Run<?,?> build);
-  
-  /**
-   * Decorates external process launcher running on a node.
-   * @param original Original launcher
-   * @param run Run, for which the decoration should be performed
-   * @param node Target node. May be {@code master} as well
-   * @return Decorated launcher or {@code original} launcher
-   */
-  @Nonnull
-  public abstract Launcher decorateLauncher(@Nonnull Launcher original, 
-          @Nonnull Run<?,?> run, @Nonnull Node node); 
-  
-  /** 
-   * Fallback Logging methods for jobs, which do not define the implementation. 
-   */
-  public static final LoggingMethod NOOP = new NoopLoggingMethod();
-  private static class NoopLoggingMethod extends LoggingMethod {
+
+    /**
+     * Decorates logging on the Jenkins master side.
+     * These filters can be also used for log redirection and multi-reporting.
+     * @param build Build to be decorated
+     * @return Log filter on the master. {@code null} if no custom implementation
+     */
+    @CheckForNull
+    public abstract ConsoleLogFilter createLoggerDecorator(Run<?,?> build);
+
+    /**
+     * Decorates external process launcher running on a node.
+     * @param original Original launcher
+     * @param run Run, for which the decoration should be performed
+     * @param node Target node. May be {@code master} as well
+     * @return Decorated launcher or {@code original} launcher
+     */
+    @Nonnull
+    public Launcher decorateLauncher(@Nonnull Launcher original,
+        @Nonnull Run<?,?> run, @Nonnull Node node) {
+        if (node instanceof Jenkins) {
+            return new DefaultLocalLauncher(original);
+        } else {
+            return new DefaultRemoteLauncher(original, run, this);
+        }
+    }
+
+    /**
+     * Provides the output stream for given run.
+     * @param run Run
+     * @return
```

Comment:
Fine for prototype, but does not work for production code. Javadoc linter returns errors by default if tags are not documented like <code>returns</code> entries